salt/netapi/rest_tornado/saltnado.py

```diff
@@ -740,7 +740,17 @@ def post(self):
 
         # Grab eauth config for the current backend for the current user
         try:
-            perms = self.application.opts['external_auth'][token['eauth']][token['name']]
+            eauth = self.application.opts['external_auth'][token['eauth']]
+            # Get sum of '*' perms, user-specific perms, and group-specific perms
+            perms = eauth.get(token['name'], [])
+            perms.extend(eauth.get('*', []))
+
+            if 'groups' in token and token['groups']:
+                user_groups = set(token['groups'])
+                eauth_groups = set([i.rstrip('%') for i in eauth.keys() if i.endswith('%')])
+
+                for group in user_groups & eauth_groups:
+                    perms.extend(eauth['{0}%'.format(group)])
```

Comment:
I see a behavior change here. If not user name nor groups are in eauth list before we were getting `send_error(401)`. Now we're returning with empty perms list.
If this change is expected I'm OK. If not it's better to detect match with a flag keeping `send_error` if not name nor group match found.

Reply:
good catch, i just copied this from rest_cherrypy. I'll restore send_error(401)

Reply:
actually on further review it looks like this is expected behavior in cherrypy. I am indifferent here, is there any policy of attempting to make saltnado compatible with cherrypy? is cherrypy the canonical implementation?

Reply:
on further thought, I think aligning with cherrypy makes sense here. the reason I hit this, and likely why anyone else would, is theyve transitioned from cherrypy with pepper and are moving to tornado. functionality should be 1 to 1 as close as possible. so working as designed :)

Reply:
I'm OK then. Thank you for your work.